policy/modules/admin/sudo.if

```diff
@@ -203,6 +204,8 @@ template(`sudo_role_template',`
 
 		ifdef(`init_systemd',`
 			init_dbus_chat($1_sudo_t)
+			systemd_dbus_chat_logind($1_sudo_t)
+			systemd_read_logind_state($1_sudo_t)
 		')
 	')
```

Comment:
I think I have updated as you asked.  I did leave the "ifdef(init_systemd)" as that was there before.  Let me know if I misunderstood or you want additional changes.

Reply:
Ideally everything for enabling systemd would depend on the presence of the systemd module. In cases like this, the `ifdef(init_systemd)` is because there wasn't anything depending on the systemd module. By adding the `systemd_*` calls, now there are dependencies.

Reply:
Ok, ifdef(init_systemd) also removed.

Reply:
Sorry, I made a mistake. Please keep the ifdef. I confused this with one of the other changes.

Reply:
replace ifdef